### PR TITLE
helps prevent netspeak in ic.

### PR DESCRIPTION
**Pull request by F-OS**
my, and probably the admins biggest pet peeve.

:cl:
rscadd: prevents netspeak memes.
/:cl:

### CODE REVIEW

**Comment by Shapsy** [comment]
aye lamo

**Comment by optimumtact** [comment]
Is this a real PR because there are some code issues, but I don't want to bother explaining them if not.

**Comment by F-OS** [comment]
@optimumtact really tired due to the time, what issues?

**Comment by F-OS** [comment]
travis just told me, nevermind, will fix and reopen.

**Comment by F-OS** [comment]
@optimumtact and yes, this is real.

**Comment by F-OS** [comment]
so tired.

**Comment by optimumtact** [comment]
Actually this is related to this:
https://www.morewords.com/contains/lol/

There are 72 legitimate words containing lol and your system will flag and punish anyone attempting to use any of them.

I.E OOC:I really want a lollypop
but now I have braindamage instead

But you should use regex instead and check for the existence of space either side of the word to be clear that it is actually lol and not something else.

In the end however this system is simply not worth it - people will simply evade the word filters with trivial modifications l  ol and so on. You end up maintaining a larger and larger blacklist of regex to manage the issue and it's simply not worth it.

Leave it up to the administrators to enforce any speech standards in ooc.

**Comment by F-OS** [comment]
yeah.

File: code/modules/mob/living/say.dm [hunk 1]
```diff
@@ -39,27 +39,31 @@ var/list/department_radio_keys = list(
...
+	  ":ï" = "changeling",	"#ï" = "changeling",	".ï" = "changeling"
 )
 
 var/list/crit_allowed_modes = list(MODE_WHISPER,MODE_CHANGELING,MODE_ALIEN)
 
 /mob/living/say(message, bubble_type,)
 	message = trim(copytext(sanitize(message), 1, MAX_MESSAGE_LEN))
+	if(findtext(message, "lol") || findtext(message, "omg") || findtext(message, "lamo") findtext(message, "ikr") || findtext(message, "brb") || findtext(message, "rotfl"))
+		adjustBrainLoss(60) // TG is a meme free zone.
+		C << 'sound/effects/adminhelp.ogg'
```
Comment (optimumtact) [inline code/modules/mob/living/say.dm]: Also who or what object is C?

Comment (F-OS) [inline code/modules/mob/living/say.dm]: again, tired. copypasted it from the adminhelp code.

**Comment by F-OS** [comment]
fixed it. and this is not ment to replace admins, it is ment to help them. if somebody does say "l o l" let the admins punish them.

**Comment by Shapsy** [comment]
but what if the admins don't care and say lol themselves

**Comment by F-OS** [comment]
@Shapsy then the admins get brain damage. https://tgstation13.org/wiki/Admin_Conduct.
Don't be a hypocrite. If you're doing something you'd normally ban someone for doing, you're breaking this.

**Comment by kevinz000** [comment]
This is bad

**Comment by F-OS** [comment]
@kevinz000 why?

**Comment by kevinz000** [comment]
Because I like memes and 60 brainloss is a lot

**Comment by Shapsy** [comment]
oh wait for some reason I thought this applied to OOC, nevermind this is fine

**Comment by F-OS** [comment]
@kevinz000 any admin would make you a retard for netspeak in ooc.

**Comment by kevinz000** [comment]
Also maybe put something that lets the victim understand why they're being censored. You realize many of the people this is intended to hit have not read the rules and putting "tg is a meme free Zone" will not help them understand and just frustrate them right?

**Comment by F-OS** [comment]
@Shapsy yeah, this is for ic only.

**Comment by F-OS** [comment]
@kevinz000 yeah, good point.

**Comment by kevinz000** [comment]
Gee man then you haven't been to sybil because I've seen people netspeaking ignored

**Comment by Shapsy** [comment]
No admin would make someone retarded for using netspeak in OOC

**Comment by F-OS** [comment]
@kevinz000 i too heav seen the horrible meme spouting on sybil.

**Comment by F-OS** [comment]
@shapsy huh.

**Comment by kevinz000** [comment]
.... reminds me of "prevents ic in ooc"
Player issue not code issue.

**Comment by Shapsy** [comment]
You said "any admin would make you a retard for netspeak in ooc." and that's not true at all

File: code/modules/mob/living/say.dm [hunk 2]
```diff
@@ -39,27 +39,31 @@ var/list/department_radio_keys = list(
...
 )
 
 var/list/crit_allowed_modes = list(MODE_WHISPER,MODE_CHANGELING,MODE_ALIEN)
 
 /mob/living/say(message, bubble_type,)
 	message = trim(copytext(sanitize(message), 1, MAX_MESSAGE_LEN))
+	if(findtext(message, "lol ") || findtext(message, "omg ") || findtext(message, "lamo ") || findtext(message, "ikr ") || findtext(message, "brb ") || findtext(message, "rotfl "))
+		adjustBrainLoss(10) // TG is a meme free zone.
+		src << 'sound/effects/adminhelp.ogg'
+		src << "<span class='red' size='2'>no net-speak in ic, TG is a meme free zone.</span>"
```
Comment (kevinz000) [inline code/modules/mob/living/say.dm]: Caps. Clarify?maybe?

File: code/modules/mob/living/say.dm [hunk 3]
```diff
@@ -39,27 +39,31 @@ var/list/department_radio_keys = list(
...
+	  ":å" = "Syndicate",	"#å" = "Syndicate",		".å" = "Syndicate",
+	  ":é" = "Supply",		"#é" = "Supply",		".é" = "Supply",
+	  ":ï" = "changeling",	"#ï" = "changeling",	".ï" = "changeling"
 )
 
 var/list/crit_allowed_modes = list(MODE_WHISPER,MODE_CHANGELING,MODE_ALIEN)
 
 /mob/living/say(message, bubble_type,)
 	message = trim(copytext(sanitize(message), 1, MAX_MESSAGE_LEN))
+	if(findtext(message, "lol ") || findtext(message, "omg ") || findtext(message, "lamo ") || findtext(message, "ikr ") || findtext(message, "brb ") || findtext(message, "rotfl "))
```
Comment (kevinz000) [inline code/modules/mob/living/say.dm]: Lmao* Rofl*

**Comment by F-OS** [comment]
@shapsy right. that was the wrong thing to say, sorry.

**Comment by RandomMarine** [comment]
As much as I'd like to see people get braindamaged by this. There are other servers running /tg/ code, the codebase shouldn't enforce this.

**Comment by Armhulen** [comment]
>join game
>not antag
>LOLLMAOROFLJKOMGROFLMAO
>tab out for forty mins
who else devilish here

File: code/modules/mob/living/say.dm (hunk 3 above)
Comment (GunHog) [inline code/modules/mob/living/say.dm]: Add "afk"

File: code/modules/mob/living/say.dm [hunk 4]
```diff
@@ -39,27 +39,31 @@ var/list/department_radio_keys = list(
...
+		adjustBrainLoss(10) // TG is a meme free zone.
+		src << 'sound/effects/adminhelp.ogg'
+		src << "<span class='red' size='2'>No net-speak in IC, TG is a meme free zone.</span>"
```
Comment (PKPenguin321) [inline code/modules/mob/living/say.dm]: This is outright untrue

**Comment by PKPenguin321** [comment]
How does this interact with forced speech from brain damage?

**Comment by RandomMarine** [comment]
Challenge mode. Code gaining 5 brainloss for each E in REEEEEEEEEEEEEEEEEE

**Comment by optimumtact** [comment]
Chiming back in here to say if this is going ahead it should be a list of regex loaded from a text file and cached into memory at round start.

This gives optimum flexibility to the server hosts in deciding what things need to be enforced/not enforced.

**Comment by Niknakflak** [comment]
RIP, purposefully speaking in meme ICly in the thunderdome

**Comment by coiax** [comment]
Does anyone remember Habbo Hotel, which censored naughty words like "sex" into the word bobba?

So, this example would become:
So, thi bobba ample would become

You think  stripping bad words is easy? And we want to give people the ability to deliberately and easily brain damage themselves? No no no. :-1:

**Comment by Bawhoppen** [comment]
:-1: for the changelog

**Comment by Supermichael777** [comment]
i for one regularly use xylene xylol every time i play chemist.

File: code/modules/mob/living/say.dm (hunk 1 above)
Comment (Supermichael777) [inline code/modules/mob/living/say.dm]: just cut the sound

**Comment by F-OS** [comment]
Ok, i have gotten 6 hours of sleep and some coffee, and now that i see this, i realize this needs TONS of work, gonna close this until i fix this.

**Comment by Hatfish** [comment]
We goon now.

**Pull request closed without merging**